agent/hcp/telemetry/otel_exporter.go

```diff
@@ -0,0 +1,53 @@
+package telemetry
+
+import (
+	"context"
+
+	hcpclient "github.com/hashicorp/consul/agent/hcp/client"
+	"github.com/hashicorp/go-multierror"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/aggregation"
+	"go.opentelemetry.io/otel/sdk/metric/metricdata"
+)
+
+// OTELExporter is a custom implementation of a OTEL Metrics SDK metrics.Exporter.
+// The exporter is used by a OTEL Metrics SDK PeriodicReader to export aggregated metrics.
+// This allows us to use a custom client - HCP authenticated MetricsClient.
+type OTELExporter struct {
+	client   hcpclient.MetricsClient
+	endpoint string
+}
+
+// Temporality returns the Cumulative temporality for metrics aggregation.
+// Telemetry Gateway stores metrics in Prometheus format, so use Cummulative aggregation as default.
+func (e *OTELExporter) Temporality(_ metric.InstrumentKind) metricdata.Temporality {
+	return metricdata.CumulativeTemporality
+}
+
+// Aggregation returns the Aggregation to use for an instrument kind.
+func (e *OTELExporter) Aggregation(kind metric.InstrumentKind) aggregation.Aggregation {
```

Comment:
We could use the default provided by the library to make this function a one-liner :  `return metric.DefaultAggregationSelector(kind)`

However, this implementation [panics](https://github.com/open-telemetry/opentelemetry-go/blob/2e54fbb3fede5b54f316b3a08eab236febd854e0/sdk/metric/reader.go#L147), which worries me a bit. So I wrote a custom simple one.

Reply:
done! thanks